lib/rdoc/markup/to_html.rb

```diff
@@ -98,7 +98,11 @@ def handle_RDOCLINK(url) # :nodoc:
 
       gen_url CGI.escapeHTML(url), CGI.escapeHTML(text)
     when /^rdoc-image:/
-      url, alt = $'.split(":", 2) # Split the string after "rdoc-image:" into url and alt
+      # Split the string after "rdoc-image:" into url and alt
+      url_alt = $'
+      split_pos = url_alt.index(%r{:[^/]})
+      url = split_pos ? url_alt[0, split_pos] : url_alt
+      alt = split_pos ? url_alt[split_pos + 1, url_alt.size] : nil
```

Comment:
It looks a bit messy, and there might be a better code here... Welcome any suggestions! 😃

Reply:
Improved via https://github.com/ruby/rdoc/pull/1348#discussion_r2060084903